browser/extensions/api/brave_action_api.cc

```diff
@@ -96,52 +96,44 @@ BraveActionAPI* BraveActionAPI::Get(Browser* context) {
 }
 
 // static
-bool BraveActionAPI::ShowActionUI(
-      ExtensionFunction* extension_function,
-      const std::string& extension_id,
-      std::unique_ptr<int> window_id_param,
-      std::unique_ptr<std::string> ui_relative_path_param,
-      std::string* error) {
+base::expected<bool, std::string> BraveActionAPI::ShowActionUI(
+    ExtensionFunction* extension_function,
+    const std::string& extension_id,
+    absl::optional<int> window_id_param,
+    absl::optional<std::string> ui_relative_path_param) {
```

Comment:
Should these be const references here and elsewhere? It seems like we do that in a few places in this PR but not everywhere. Not sure what the best practice is for `absl::optional`, but would be good to be consistent.

Reply:
There's a lot of issues with `const optional<T>&` because they are not an optional reference, but a reference to a const optional, and there for they are not encouraged in chromium. It is a subtlety that can lead to quite surprising misguided assumptions. The reason I had specifically used the `const optional<T>&` in the other particular case is that it would prevent the use-after-move there, even if reintroduced by accident, as moving from a `const optional<T>&` will always result in a copy.

If you prefer, for the sake of consistency, I can make all these arguments `const optional<T>&`, as it will be of no consequence.

Reply:
Ah OK, I see. Fine by me as-is then!